sympy/solvers/solvers.py

```diff
@@ -847,13 +848,24 @@ def solve(f, *symbols, **flags):
     def _sympified_list(w):
         return list(map(sympify, w if iterable(w) else [w]))
     bare_f = not iterable(f)
+
+    # check flag usage for particular/quick which should only be used
+    # with systems of equations
+    if flags.get('quick', None) is not None:
+        if not flags.get('particular', None):
+            raise ValueError('when using `quick`, `particular` should be True')
```

Comment:
A warning would be better than an exception here. If someone is using this mistakenly then they can fix their code after seeing the warning but their code will continue to work in the mean time.